java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -1759,21 +1676,17 @@ public ArgumentListTreeImpl completeArguments(InternalSyntaxToken openParenthesi
       new ArgumentListTreeImpl(openParenthesisToken, closeParenthesisToken);
   }
 
-  public ArgumentListTreeImpl newArguments(ExpressionTree expression, Optional<List<AstNode>> rests) {
-    List<AstNode> children = Lists.newArrayList();
+  public ArgumentListTreeImpl newArguments(ExpressionTree expression, Optional<List<Tuple<InternalSyntaxToken, ExpressionTree>>> rests) {
     ImmutableList.Builder<ExpressionTree> expressions = ImmutableList.builder();
-
-    children.add((AstNode) expression);
     expressions.add(expression);
-
     if (rests.isPresent()) {
-      for (AstNode rest : rests.get()) {
-        children.addAll(rest.getChildren());
-        expressions.add((ExpressionTree) rest.getLastChild());
+      for (Tuple<InternalSyntaxToken, ExpressionTree> rest : rests.get()) {
+        //FIXME : SONARJAVA-547 Comma should be handled : rest.first()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)